lib/semantics/resolve-labels.cc

```diff
@@ -0,0 +1,969 @@
+// Copyright (c) 2018, NVIDIA CORPORATION.  All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "resolve-labels.h"
+#include "../common/enum-set.h"
+#include "../parser/message.h"
+#include "../parser/parse-tree-visitor.h"
+#include <cassert>
+#include <cctype>
+#include <cstdarg>
+#include <iostream>
+
+namespace {
+
+using namespace Fortran;
+using namespace parser::literals;
+
+ENUM_CLASS(TargetStatementEnum, Do, Branch, Format)
+using TargetStmtType = common::EnumSet<TargetStatementEnum, 8>;
+
+using IndexList = std::vector<std::pair<parser::CharBlock, parser::CharBlock>>;
+using ScopeProxy = unsigned;
```

Comment:
What is a `ScopeProxy`? I can't tell from the type so a comment would be helpful.

Reply:
added comment